rawx-apache2/src/rawx_repo_core.c

```diff
@@ -445,6 +445,51 @@ check_chunk_info_with_trailers(apr_pool_t *pool,
 	return NULL;
 }
 
+apr_status_t
+chunk_verify_checksum(dav_resource *resource, request_rec *r)
+{
+	apr_file_t *fd = NULL;
+	apr_status_t status = apr_file_open(&fd, resource_get_pathname(resource),
+		APR_FOPEN_READ|
+		APR_FOPEN_BINARY,
+		0, resource->pool);
+	if (status != APR_SUCCESS) {
+		/* It should be already catched by resource_stat_chunk */
+		return status;
+	}
+
+	void *buf = apr_palloc(resource->pool, DEFAULT_BLOCK_SIZE);
+	apr_size_t nbytes = DEFAULT_BLOCK_SIZE;
+
+	GChecksum *md5 = g_checksum_new(G_CHECKSUM_MD5);
+	while ( (status = apr_file_read(fd, buf, &nbytes)) == APR_SUCCESS) {
+		g_checksum_update(md5, buf, nbytes);
+		nbytes = DEFAULT_BLOCK_SIZE;
+	}
+	apr_file_close(fd);
+
+	if (status != APR_EOF) {
+		return status;
+	}
+
+	/* use hash from x-oio-chunk-meta-chunk-hash or from xatter ?*/
+	const char *hash_to_verify = NULL;
+	if ( !(hash_to_verify = apr_table_get(r->headers_in, "x-oio-chunk-meta-chunk-hash" )) ) {
```

Comment:
I suggest defining a constant for `"x-oio-chunk-meta-chunk-hash"`, derived from `RAWX_HEADER_PREFIX` in `core/oiocfg.h`.